### PR TITLE
feat new at_acoustic pipeline

### PR DESCRIPTION
**Pull request by evacougnon**
Need this [PR](https://github.com/aodn/python-aodncore/pull/267) to be merged before the test can pass

### CODE REVIEW

File: aodndata/aatams/aatams_acoustic.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+AT_ACOUSTIC_REGEX = '^IMOS_ATF-ACOUSTIC_TAGID_A(69|180)-[0-9]{4}-[0-9]{3,5})_[0-9]{1,10}_[0-9]{1,10})*\.csv$'
+# AT_ACOUSTIC_REGEX_DAILY
+FILE_TYPE_NEED_INDEX = ('QC', 'QC_daily_summary')
+AT_ACOUSTIC_FILE_PATTERN = re.compile(r"""
+                                      ^IMOS_ATF-ACOUSTIC_TAGID_
+                                      (?P<transmitter_id>^A(69|180)-[0-9]{4}-[0-9]{3,5})_
+                                      (?P<release_id>[0-9]{1,10}_[0-9]{1,10})
+                                      (?P<product_type>.*)
+                                      \.csv$""", re.VERBOSE)
```
[the remark this answers is not in the document]
Comment (mhidas) [reply aodndata/aatams/aatams_acoustic.py:30]: Also, if you want the optional `product_type` field to be separated from the rest by a "_", the last line should be
```regexp
    (_(?P<product_type>.+))?
```

**Review by mhidas — COMMENTED** [review]
I think there was only one or two really significant changes, the rest are just suggestions to simplify.
If it's doing what you need it to do, it's all good!

File: aodndata/aatams/aatams_acoustic.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (evacougnon) [reply aodndata/aatams/aatams_acoustic.py:30]: well spotted for the `^` thanks

About the product type, there won't always be a `_product`. Not including the `_` allows anything to be the product type and can be null

[the remark this answers is not in the document]
Comment (mhidas) [reply aodndata/aatams/aatams_acoustic.py:30]: :+1: Ok (but if there is a "_" it will be matched as part of the `product_type` field)

**Comment by evacougnon** [comment]
Note for myself: update dest_path

File: aodndata/aatams/aatams_acoustic_index_tmp.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+           first_indexed: boolean,
+                          True: has already been harvested and about to harvest a newer version
+                          False: first to indexing this file
+                          optional
+    :return: dataframe with file metadata
+    """
+
+    df_csv_extract = create_empty_dataframe()
+    # df_csv = pd.read_csv(file2read.local_path)  # loads the entire csv in memory
+    df_csv_1 = pd.read_csv(input_file.local_path, nrows=1)  # loads only the first row of data
```
[the remark this answers is not in the document]
Comment (evacougnon) [reply aodndata/aatams/aatams_acoustic_index_tmp.py:55]: For some reasons it doesn't like it, I get an empty dataframe

[the remark this answers is not in the document]
Comment (mhidas) [reply aodndata/aatams/aatams_acoustic_index_tmp.py:55]: Oops, I think that's actually the wrong method. Try `df_csv.iloc[0]` (see [here](https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#different-choices-for-indexing) for an explanation of the difference)

File: aodndata/aatams/aatams_acoustic.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+        tag_files_list = self.file_collection.filter_by_attribute_regex('name', r'^IMOS_ATF-ACOUSTIC_TAGID_.*\.csv$')
+        index_tmp_file_path = os.path.join(self.temp_dir, 'index_tmp.csv')
+        df_index_tmp = create_empty_dataframe()
+        for tag_file in tag_files_list:
+            # Check if each csv file have already been harvested and exist on S3 (with same name)
+            # -if yes: the csv files has already been indexed, do not create a first_indexed date
+            # -if not: new record, include a first_indexed date in index_tmp.csv
+            destination_s3 = self.dest_path(tag_file.name)
+            storage_query_res = self.state_query.query_storage(destination_s3).keys()
```
[the remark this answers is not in the document]
Comment (mhidas) [reply aodndata/aatams/aatams_acoustic.py:117]: :+1:  that's fine, and I don't need to convince you to change your mind, but... 
* Just be aware that you're making an assumption here: That a file existing on S3 implies it has been indexed into the database. This _should_ be true if the only way any of these files ever got to S3 is via your handler in its current form (but we've certainly seen cases of S3 and database content getting out of sync...)
* I would also check that the timestamps are properly converted from Pandas DataFrame to CSV to DB.

[the remark this answers is not in the document]
Comment (evacougnon) [reply aodndata/aatams/aatams_acoustic.py:117]: That's a very good point! I didn't think about this scenario, I'll update the pipeline

File: aodndata/aatams/aatams_acoustic_index_tmp.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (evacougnon) [reply aodndata/aatams/aatams_acoustic_index_tmp.py:55]: For some reason it still gives me an empty dataframe, only `values[0]` seems to work but then I have to read only one row

**Review by mhidas — APPROVED** [review]
:+1: Just a couple of minor suggestions, but good to go!

**Pull request merged**